Making ops c10 full: ops returning multiple out arguments

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49259 Remove `use_c10_dispatcher: full` lines
* #49251 Remove generated_unboxing_wrappers and setManuallyBoxedKernel
* #49220 Remove .impl_UNBOXED() and functionalities associated with it
* #49164 Remove codegen logic to support non-c10-full ops
* #49138 [wip] Making ops c10-full: list of optional tensors
* #49150 Making ops c10-full: ConstQuantizerPtr
* #49146 Making ops c10-full: Storage arguments
* #49088 Making ops c10-full: optional lists
* #49083 Making ops c10 full: optional out arguments
* #49013 Making ops c10-full: Generator arguments
* #49012 Making ops c10 full: out overloads with default arguments
* #49008 Making ops c10-full: Dimname arguments
* #49007 Making ops c10-full: ops blocked by manual registrations
* **#49006 Making ops c10 full: ops returning multiple out arguments**
* #48912 Make out ops c10-full (with hacky-wrapper)

There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

smessmer committed:
Making ops c10 full: ops returning multiple out arguments

There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -269,6 +269,7 @@
   variants: method
 
 - func: absolute.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
+  use_c10_dispatcher: hacky_wrapper_for_legacy_signatures
```
Why wouldn't all these single-return ops go in the preceding PR?

bhosmer replied:
Ah, reread the PR description. It would be nice to move these to the previous PR for hygiene, I think - shouldn't be too hard, right?

bhosmer reviewed (approved):
Are there tests for make_boxed_from_unboxed's multiple-returns codepath(s) already?

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]

smessmer committed:
Update on "Making ops c10 full: ops returning multiple out arguments"


There was an issue in the unboxing logic with ops returning multiple out arguments.
This PR fixes that and makes those ops c10 full.

Additionally, it makes some ops c10 full that slipped through the cracks before.

Differential Revision: [D25392592](https://our.internmc.facebook.com/intern/diff/D25392592/)

[ghstack-poisoned]
